packages/shared/tests/normalizeRoutePath.spec.ts

```diff
@@ -15,55 +15,168 @@ const testCases = [
   ['/foo/index.md', '/foo/'],
   ['/foo/index.html', '/foo/'],
   ['/foo/index', '/foo/'],
-  ['', ''],
   ['README.md', 'index.html'],
   ['readme.md', 'index.html'],
   ['index.md', 'index.html'],
   ['index.html', 'index.html'],
   ['index', 'index.html'],
-  ['foo/', 'foo/'],
-  ['foo/README.md', 'foo/'],
-  ['foo/readme.md', 'foo/'],
-  ['foo/index.md', 'foo/'],
-  ['foo/index.html', 'foo/'],
-  ['foo/index', 'foo/'],
 
-  // non-index
+  // absolute non-index
   ['/foo', '/foo.html'],
   ['/foo.md', '/foo.html'],
   ['/foo.html', '/foo.html'],
   ['/foo/bar', '/foo/bar.html'],
   ['/foo/bar.md', '/foo/bar.html'],
   ['/foo/bar.html', '/foo/bar.html'],
+
+  // relative index without current
+  ['foo/', 'foo/'],
+  ['foo/README.md', 'foo/'],
+  ['foo/readme.md', 'foo/'],
+  ['foo/index.md', 'foo/'],
+  ['foo/index.html', 'foo/'],
+  ['foo/index', 'foo/'],
+
+  // relative non index without current
   ['foo', 'foo.html'],
   ['foo.md', 'foo.html'],
   ['foo.html', 'foo.html'],
   ['foo/bar', 'foo/bar.html'],
   ['foo/bar.md', 'foo/bar.html'],
   ['foo/bar.html', 'foo/bar.html'],
 
-  // hash and query
-  ['/foo#bar', '/foo.html#bar'],
-  ['/foo.md#bar', '/foo.html#bar'],
-  ['/foo.html#bar', '/foo.html#bar'],
-  ['/foo?bar=baz', '/foo.html?bar=baz'],
-  ['/foo.md?bar=baz', '/foo.html?bar=baz'],
-  ['/foo.html?bar=baz', '/foo.html?bar=baz'],
-  ['/foo?bar=baz#qux', '/foo.html?bar=baz#qux'],
-  ['/foo.md?bar=baz#qux', '/foo.html?bar=baz#qux'],
-  ['/foo.html?bar=baz#qux', '/foo.html?bar=baz#qux'],
-  ['foo#bar', 'foo.html#bar'],
-  ['foo.md#bar', 'foo.html#bar'],
-  ['foo.html#bar', 'foo.html#bar'],
-  ['foo?bar=baz', 'foo.html?bar=baz'],
-  ['foo.md?bar=baz', 'foo.html?bar=baz'],
-  ['foo.html?bar=baz', 'foo.html?bar=baz'],
-  ['foo?bar=baz#qux', 'foo.html?bar=baz#qux'],
-  ['foo.md?bar=baz#qux', 'foo.html?bar=baz#qux'],
-  ['foo.html?bar=baz#qux', 'foo.html?bar=baz#qux'],
-  ['#bar', '#bar'],
-  ['?bar=baz', '?bar=baz'],
-  ['?bar=baz#qux', '?bar=baz#qux'],
+  // relative non index with current
+  ['foo', '/foo.html', '/'],
```

Comment:
Better to use `[[input1, input2], output]` format, instead of putting the second param  as the third item of the tuple